Exiled.API/Features/Player.cs

```diff
@@ -2467,25 +2461,16 @@ public bool TryAddCandy(CandyKindID candyType)
         {
             if (Scp330Bag.TryGetBag(ReferenceHub, out Scp330Bag bag))
             {
-                bool flag = bag.TryAddSpecific(candyType);
-
-                if (flag)
-                    bag.ServerRefreshBag();
-
-                return flag;
+                return ((Scp330)Item.Get(bag)).AddCandy(candyType);
             }
 
             if (Items.Count > 7)
                 return false;
 
             Scp330 scp330 = (Scp330)AddItem(ItemType.SCP330);
 
-            Timing.CallDelayed(0.02f, () =>
-            {
-                scp330.Base.Candies.Clear();
-                scp330.AddCandy(candyType);
```

Comment:
1-frame delay is necessary to give client time to create the bag object on it's end before trying to add something to it.